app/com/linkedin/drelephant/tuning/engine/SparkHBTParamRecommender.java

```diff
@@ -360,4 +430,62 @@ private int getMemoryIncreaseForGC() {
       return 0;
     }
   }
+
+  /**
+   * This method suggest three parameters : spark.dynamicAllocation.minExecutors, spark.dynamicAllocation.maxExecutors
+   *  and spark.executor.instances
+   * For now we are changing these parameters only in case we are changing number of executor core
+   */
+  private void suggestNumExecutorParameters() {
+    if (suggestedCore != lastRunExecutorCore) {
+      if (lastRunDynamicAllocationMinExecutors != null) {
+        suggestedDynamicAllocationMinExecutors =
+            (int) Math.ceil(lastRunDynamicAllocationMinExecutors * lastRunExecutorCore / suggestedCore);
+      }
+      if (lastRunDynamicAllocationMaxExecutors != null) {
```

Comment:
Is it necessary to set max executors? Looking at our internal spark code, it seems to be calculated as under:
Seq(Seq(numExecutors, initialExecutors).max, maxNumExecutors).min

Which means min ( max(executor instances, initial executors), max executors))

Reply:
I think , this is internal to linkedin change .

Reply:
As discussed for now we are handling it by suggesting only one parameter numExecutors. We would need to think about it more, how we will integrate it in open source.